builtin/logical/aws/path_static_roles.go

```diff
@@ -219,6 +219,10 @@ func (b *backend) pathStaticRolesWrite(ctx context.Context, req *logical.Request
 		if err != nil {
 			return nil, fmt.Errorf("expected an item with name %q, but got an error: %w", config.Name, err)
 		}
+		// check if i is nil to prevent panic
+		if i == nil {
```

Comment:
PopByKey documents that it will return nil if the item is not found. So i think this nil check is required. However, I am wondering if we should also update PopByKey to return queue.ErrEmpty when the queue is empty?

Reply:
Not a bad idea. I was thinking for returning the error here in the second `nil` space, but would need to understand any upstream effects:

```
	item, ok := pq.dataMap[key]
	if !ok {
		return nil, nil
	}
```

Thoughts?

Reply:
Above code is from the PopByKey definition in the SDK.

Reply:
I have given it some more thought and I don't think we should change the SDK behavior in this PR since it is out of scope for this fix.